Changes to how test creates and connects to pool.

amjustin13 opened the pull request:
Removed the fixed gid and uid values from the yml file and added the function call to pool_create from general_utils.py
Signed-off-by: Amanda Justiniano <amanda.justiniano-pagn@intel.com>

daosbuild1 commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -29,8 +29,9 @@
 
-import agent_utils
-import server_utils
+import agent_utils import run_agent, stop_agent
```
(pylint-syntax-error) invalid syntax

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-676/1/execution/node/44/log

daosbuild1 commented:
Test stage Build RPM on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-676/1/execution/node/227/log

phender commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -51,48 +52,32 @@ def test_container_basics(self):
-            hostlist = self.params.get("test_machines", '/run/hosts/*')
-            hostfile = write_host_file.write_host_file(hostlist,
-                                                       self.workdir)
+            self.hostlist_servers = self.params.get("test_machines", '/run/hosts/*')
```
Lines 58-63 could be removed if you changed the inheritance to `TestWithServers`.  You could also remove lines 46-48 and 105-107 with this change.

daosbuild1 replied:
(pylint-line-too-long) Line too long (84/80)

phender commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -51,48 +52,32 @@ def test_container_basics(self):
         just making sure they work at a rudimentary level
 
-        :avocado: tags=container,containercreate,containerdestroy,basecont
+        :avocado: tags=pr,container,containercreate,containerdestroy,basecont
```
Also add the `all` tag

amjustin13 replied:
Will add

phender commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -51,48 +52,32 @@ def test_container_basics(self):
-            hostlist = self.params.get("test_machines", '/run/hosts/*')
-            hostfile = write_host_file.write_host_file(hostlist,
-                                                       self.workdir)
+            self.hostlist_servers = self.params.get("test_machines", '/run/hosts/*')
```
Does this yield the single host list (`test_machines1`) or the list of 6 servers (`test_machines6`) from the yaml file.  I do not see a reference to either path in the original code (either `/run/hosts/test_amchines1/test_machines` or `/run/hosts/test_amchines6/test_machines`).  If only one server is required can we cleanup the yaml file?

amjustin13 replied:
The yaml is using a !mux on the 'hosts' tag. So it should have a pass with one machine and a pass with 6 machines. I am not sure what the purpose of have 6 machines is, we'd have to verify the intent with the original author. Since this test is for basic container (create, destroy, open, query) functions, I have no issue cleaning up the yaml to only have one host.

saurabhtandan commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -51,48 +52,32 @@ def test_container_basics(self):
+            run_server(self.hostfile_servers, self.server_group, self.basepath)
 
             # give it time to start
             time.sleep(2)
```
We can also remove sleep() here. server_utils.py takes care if servers are started properly.

amjustin13 replied:
Will remove.

phender commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -51,48 +52,32 @@ def test_container_basics(self):
-            hostlist = self.params.get("test_machines", '/run/hosts/*')
-            hostfile = write_host_file.write_host_file(hostlist,
-                                                       self.workdir)
+            self.hostlist_servers = self.params.get("test_machines", '/run/hosts/*')
```
Lines 58-63 could be removed if you changed the inheritance to TestWithServers. You could also remove lines 46-48 and 105-107 with this change.

amjustin13 replied:
Good catch, I'll make that change.

daosbuild1 commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -51,48 +52,32 @@ def test_container_basics(self):
-            container.create(pool.handle)
+            # Create pool and connect
+            self.pool = get_pool(
+            self.context, pool_mode, pool_size, pool_name, 1, self.d_log)
```
(pylint-bad-continuation) Wrong hanging indentation (add 4 spaces).

daosbuild1 commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -51,48 +52,32 @@ def test_container_basics(self):
-                                                       self.workdir)
+            self.hostlist_servers = self.params.get("test_machines", '/run/hosts/*')
+            self.hostfile_servers = write_host_file.write_host_file(
+                                    self.hostlist_servers, self.workdir)
```
(pylint-bad-continuation) Wrong hanging indentation (remove 20 spaces).

daosbuild1 commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -51,48 +52,32 @@ def test_container_basics(self):
 
-            self.agent_sessions = agent_utils.run_agent(self.basepath, hostlist)
-            server_utils.run_server(hostfile, self.server_group, self.basepath)
+            self.agent_sessions = run_agent(self.basepath, self.hostlist_servers)
```
(pylint-line-too-long) Line too long (81/80)

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-676/2/execution/node/44/log

daosbuild1 commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -28,14 +28,14 @@
-import agent_utils
-import server_utils
 import write_host_file
+from agent_utils import run_agent, stop_agent
```
(pylint-unused-import) Unused stop_agent imported from agent_utils
(pylint-unused-import) Unused run_agent imported from agent_utils

daosbuild1 commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -28,14 +28,14 @@
-import server_utils
 import write_host_file
+from agent_utils import run_agent, stop_agent
+from server_utils import run_server, stop_server
```
(pylint-unused-import) Unused run_server imported from server_utils
(pylint-unused-import) Unused stop_server imported from server_utils

daosbuild1 commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -48,63 +48,36 @@ def __init__(self, *args, **kwargs):
-            # now open it
-            container.open()
+            # Create a container and open
+            self.container = get_container(self.context, self.pool, self.d_log)
```
(pylint-attribute-defined-outside-init) Attribute 'container' defined outside __init__

daosbuild1 commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -28,14 +28,14 @@
 from conversion import c_uuid_to_str
 
-class SimpleCreateDeleteTest(TestWithoutServers):
+class SimpleCreateDeleteTest(TestWithServers):
```
(pylint-undefined-variable) Undefined variable 'TestWithServers'
(pylint-too-few-public-methods) Too few public methods (1/2)

daosbuild1 commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -48,63 +48,36 @@ def __init__(self, *args, **kwargs):
-            if container.get_uuid_str() != c_uuid_to_str(
-                    container.info.ci_uuid):
+            if self.container.get_uuid_str() != c_uuid_to_str(
+                container.info.ci_uuid):
```
(pylint-bad-continuation) Wrong hanging indentation before block (add 4 spaces).
(pylint-undefined-variable) Undefined variable 'container'

daosbuild1 commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -48,63 +48,36 @@ def __init__(self, *args, **kwargs):
+        Test basic container create/destroy/open/close/query.
 
-        :avocado: tags=container,containercreate,containerdestroy,basecont
+        :avocado: tags=all,pr,container,containercreate,containerdestroy,basecont
```
(pylint-line-too-long) Line too long (81/80)

daosbuild1 commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -48,63 +48,36 @@ def __init__(self, *args, **kwargs):
-            container = DaosContainer(self.context)
-            container.create(pool.handle)
+            # Create pool and connect
+            self.pool = get_pool(
```
(pylint-attribute-defined-outside-init) Attribute 'pool' defined outside __init__

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-676/3/execution/node/45/log

daosbuild1 commented on src/tests/ftest/container/simple_create_delete_test.py:
```diff
@@ -48,63 +46,36 @@ def __init__(self, *args, **kwargs):
+        Test basic container create/destroy/open/close/query.
 
-        :avocado: tags=container,containercreate,containerdestroy,basecont
+        :avocado: tags=all,pr,container,containercreate,containerdestroy,basecont
```
(pylint-line-too-long) Line too long (81/80)

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-676/4/execution/node/45/log

jolivier23 commented:
Is there a ticket for this?

Commit message format should be
DAOS-XXX [category]: [title]
[blank line]
description

I can fix the title when landing if you just give me the text you want

amjustin13 commented:
There is a ticket:https://jira.hpdd.intel.com/browse/DAOS-2744

DAOS-2744 Container Tests: Update container test: simple_create_delete.py

Modifying test to properly populate gid and uid values. Removed the fixed gid and uid values from the yml file and added the function call to pool_create from general_utils.py
